db/node_jobs.go

```diff
@@ -154,21 +163,47 @@ func CreateBLSToExecutionChangesNodeJob(nj *types.NodeJob) (*types.NodeJob, erro
 		withdrawalCredentials := ethutil.SHA256(op.Message.FromBLSPubkey[:])
 		withdrawalCredentials[0] = byte(0) // BLS_WITHDRAWAL_PREFIX
 		if !bytes.Equal(withdrawalCredentials, v.WithdrawalCredentials) {
-			return nil, fmt.Errorf("message.FromBLSPubkey != validator.WithdrawalCredentials for validator with index %v", v.Index)
+			return nil, CreateNodeJobUserError{Message: fmt.Sprintf("fromBLSPubkey do not match withdrawalCredentials for validator with index %v", v.Index)}
 		}
 		if v.WithdrawalCredentials[0] != 0 {
-			return nil, fmt.Errorf("validator.WithdrawalCredentials[0] != 0 for validator with index %v", v.Index)
+			return nil, CreateNodeJobUserError{Message: fmt.Sprintf("withdrawalCredentials[0] != 0 for validator with index %v", v.Index)}
 		}
 		delete(opsToCheck, v.Index)
 	}
 	if len(opsToCheck) > 0 {
 		return nil, fmt.Errorf("could not check all validators")
 	}
 
-	_, err = WriterDb.Exec(`insert into node_jobs (id, type, status, data, created_time) values ($1, $2, $3, $4, now())`, nj.ID, nj.Type, nj.Status, nj.RawData)
+	tx, err := WriterDb.Beginx()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error starting db transactions: %w", err)
+	}
+	defer tx.Rollback()
+
+	for _, idx := range indicesArr {
+		res, err := tx.Exec(`insert into node_jobs_bls_changes_validators (validatorindex, node_job_id) values($1, $2) on conflict do nothing`, idx, nj.ID)
```

Comment:
This will cause several thousand queries to be executed sequentially. Better to use proper batching here.

Reply:
fixed in new e937dab